### ISSUE TITLE
Issue with some choco packages eg. Cmder

### ISSUE THREAD

**Issue by BenAffleckIsBatman** [OneGet/oneget#277]
Hi all, 

anyone able to install **cmder** from choco provider using "Install-Package cmder -force" ? 
This one tells me: 
Install-Package : Unable to create specified folder '«0»'. «» «Path cannot be the empty string or all whitespace.»

Debug tells me this happens in ProviderService::CreateFolder

I am confused by all the choco providers & prototypes out there, that's why I'm directly raising the question here.

**Comment by gaelcolas** [comment]
The short story (as I understand it) is that the original Chocolatey provider was only an [open prototype](https://github.com/chocolatey/chocolatey-oneget) independent from the Chocolatey.exe (the real tool) created a while ago.
In the meantime, Chocolatey.exe evolved, and added more features to simplify Packages.
As the Chocolatey Provider (aka. ChocolateyPrototype.exe) did not get updated along, the packages using the new features Chocolatey.exe provides, fails when you try to install using the Chocolatey Provider.

To alleviate that problem, @jianyunt created the ChocolateyGet provider (or should I say Meta-Provider?) that interface with the 'real' Chocolatey.exe (which is required) instead of using the Chocolatey Prototype.

In your case, you could Install ChocolateyGet, and use it to install cmder:
```PowerShell
PS C:\> install-packageprovider ChocolateyGet -Force

Name                           Version          Source           Summary
----                           -------          ------           -------
ChocolateyGet                  1.0.0.1          PSGallery        An PowerShell OneGet provider that disc...

PS C:\> install-package cmder -ProviderName ChocolateyGet

Choco.exe is required to continue
ChocolateyGet is built on Choco.exe. Do you want ChocolateyGet to install Choco.exe from
'https://chocolatey.org/install.ps1' now?
[Y] Yes  [N] No  [S] Suspend  [?] Help (default is "Y"): Y
PS C:\> get-package cmder

Name                           Version          Source                           ProviderName
----                           -------          ------                           ------------
cmder                          1.3.2            https://www.chocolatey.org       ChocolateyGet
```

/cc @ferventcoder who might correct me

**Comment by ferventcoder** [comment]
@gaelcolas a couple of small corrections but that is pretty much spot on:

* choco.exe (not chocolatey.exe, although there is a shim by that name)
* The prototype was not even fully featured when it was created:
   * Only One Source - it did not allow for making multiple sources, only the community repo (eliminating a large portion of the user base who use Chocolatey for organizational purposes). Multiple sources have existed since the creation of Chocolatey over six years ago.
   * Extensions (what is breaking this package in PackageManagement) - extensions existed in 2013, however it was kept as a dark feature. There was one package on the community repo that took advantage of this before it was documented (https://chocolatey.org/packages/Devbox-Common.extension, published 4/22/2013).
   * Other things I can't think of now

Folks from Microsoft that were originally set up to help moved to other teams. This is the issue to watch to know when the official provider will become available - https://github.com/chocolatey/chocolatey-oneget/issues/5

**Comment by BenAffleckIsBatman** [comment]
@ferventcoder @gaelcolas Thank you guys! I switched to the ChocolateyGet Provider which is not perfect, but works with some tweaks. Unfortunately the ChocolateyGet Provider is not optimized for work behind proxy, but could be fixed easily. I've opened a issue. 

For now a wrapped installation of Notepad, WinRAR & KeePass using ChocolateyGet within a PowerShell DSC configuration works. 
Hope to see an official provider soon - honestly its been a long way for all of us regarding these package providers. 

Thanks again!
